api/rest/init.go

```diff
@@ -56,108 +57,122 @@ var (
+		Router.MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
+			// If the host header indicates the request is going straight to
+			// the app, consider it going to the UI, direct them to it.
+			if req.Method == http.MethodGet && req.URL.Path == "/ws" && strings.HasSuffix(req.Host, fmt.Sprintf("%d", configure.Current.TracyServer.Port)) {
+				m.Route = ws
+				m.Handler = http.HandlerFunc(WebSocket)
+				m.MatchErr = nil
+
+				return true
+			}
+
+			return false
+		})
+
+		var h http.Handler
+		if v := flag.Lookup("test.v"); v != nil || configure.Current.DebugUI {
+			h = http.FileServer(http.Dir("./api/view/build"))
+		} else {
+			h = http.FileServer(assetFS())
 		}
 
-		return false
-	})
+		thost := Router.NewRoute().Name("tracyHost").BuildOnly()
+		Router.MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
+			if strings.HasPrefix(req.Host, "tracy") {
+				m.Route = thost
+				m.Handler = h
+				m.MatchErr = nil
+				return true
+			}
+			return false
+		})
 
-	var h http.Handler
-	if v := flag.Lookup("test.v"); v != nil || configure.Current.DebugUI {
-		h = http.FileServer(http.Dir("./api/view/build"))
-	} else {
-		h = http.FileServer(assetFS())
 	}
 
-	// Catch everything else.
-	t, u, d, bp, bufp := configure.ProxyServer()
-	p := proxy.New(t, u, d, bp, bufp)
+	if mode <= FULL {
 
-	thost := Router.NewRoute().Name("tracyHost").BuildOnly()
-	Router.MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
-		if strings.HasPrefix(req.Host, "tracy") {
-			m.Route = thost
-			m.Handler = h
-			m.MatchErr = nil
-			return true
-		}
-		return false
-	})
+		// Catch everything else.
+		t, u, d, bp, bufp := configure.ProxyServer()
+		p := proxy.New(t, u, d, bp, bufp)
 
-	// For CONNECT requests, the path will be an absolute URL
-	Router.SkipClean(true)
+		// For CONNECT requests, the path will be an absolute URL
+		Router.SkipClean(true)
 
-	// Proxy catches everything, except for requests back to itself.
-	prox := Router.NewRoute().Name("proxy").BuildOnly()
-	Router.MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
-		server, err := configure.ParseServer(req.Host)
-		if err != nil {
-			log.Error.Print(err)
-			return false
-		}
-		//SANITY CHECK: if we are about to send a request back to the proxy,
-		// we hit a recursion and something is up.
-		if req.Method != http.MethodConnect && server.Equal(configure.Current.TracyServer) {
-			return false
-		}
+		// Proxy catches everything, except for requests back to itself.
+		prox := Router.NewRoute().Name("proxy").BuildOnly()
+		Router.MatcherFunc(func(req *http.Request, m *mux.RouteMatch) bool {
+			server, err := configure.ParseServer(req.Host)
+			if err != nil {
+				log.Error.Print(err)
+				return false
+			}
+			//SANITY CHECK: if we are about to send a request back to the proxy,
+			// we hit a recursion and something is up.
+			if req.Method != http.MethodConnect && server.Equal(configure.Current.TracyServer) && mode != PROXY_ONLY {
```

Comment:
What is with the last conditional here? If the mode is in PROXY_ONLY mode, don't we still want to prevent a recursion. Or are you trying to say that we only need to really worry about recursions when there is an API instantiated and if there isn't an API, there shouldn't be any recursions?

Reply:
No, the api then has to go though the proxy again. If we leave this in it will not. This will cause nothing to happen when calling the api.

Reply:
I think I am being stupid, but I am still confused by this. Maybe we can chat about it in person this week. Why does the API go through the proxy?